### ISSUE TITLE
Unset XDG_RUNTIME_DIR or XDG_CONFIG_HOME silently fails

### ISSUE THREAD

**Issue by linde12** [leftwm/leftwm#970]
### Describe the enhancement

*Background:*
I've been trying out leftwm and one of the issues i had was that XDG_RUNTIME_DIR and XDG_CONFIG_HOME were not set on my system (no session manager, been using awesome single log on)

*Current behavior:*
LeftWM will silently exit with a successful status code and without any error or warning explaining why. X will then continually try to start LeftWM over and over again.

*Expected behavior:*
XDG_CONFIG_HOME to be defaulted to `˜/.config` and XDG_RUNTIME_DIR to be created/defaulted alternatively non-zero exit code with some error message and logs.

Far from everyone is using a session manager, and WMs like awesome, i3 and bspwm still run fine without a session manger so i would expect LeftWM to be fine as well.

### How important is this enhancement to you?

My workflow is impaired without it

### How important do you think is this enhancement for the average LeftWM user?

Their workflow is likely impaired without it

**Comment by 0323pin** [comment]
Actually, you don't need a session manager. I'm also running LeftWM on Void and launching X from `.bash_profile` after login from tty. It's enough to have the `dbus` service enabled and `elogind` installed.

Now, if you want the same without `elogind`, you can do it like I do on my NetBSD system and create the directory at the start of your session.
From my `.xinitrc`:

```
mkdir /tmp/${USER}-runtime && chmod -R 0700 /tmp/${USER}-runtime
export XDG_RUNTIME_DIR=/tmp/${USER}-runtime
```

**Comment by linde12** [comment]
Yes, this is what i ended up doing, but i dont see this as a solution. It would be better if LeftWM sets up whatever it needs to work for you rather than the user doing guesswork.

Regarding XDG_CONFIG_HOME i had issues with this too, but if youre telling me its defaulted it mustve been something else🤔

Anyway, XDG_RUNTIME_DIR is still something i would expect to give a clear error and non-zero exit code or just be given a sane default. The guesswork is the issue, not actually writing the line to set it once youve figured it out.

**Comment by 0323pin** [comment]
@linde12 Fair enough. Although, there's `leftwm-check`,

```
> leftwm-check
:: LeftWM version: 0.4.0
:: LeftWM git hash: c46461b6c3
:: Enabled features: lefthk lefthk-core syslog-tracing sys-log
:: Checking feature dependencies . . .
    -> lefthk OK
:: Loading configuration . . .
    -> Configuration loaded OK
:: Checking keybinds . . .
    -> All keybinds OK
:: Checking environment . . .
    -> Environment OK (has XDG_RUNTIME_DIR)
:: Checking theme . . .
    -> Theme OK
```

**Comment by VuiMuich** [comment]
Yeah, non-zero exit-code seems like a good idea, also giving a meaningfull log if this path is missing.

I have been holding back on some refactors that might be helpfull here (conforming all file handling into a utility module) as one potentially new contributor showed interest in this as well. But maybe if they don't pick it up in the next days I'll tend to this.
